### ISSUE TITLE
[1.2.0 Beta 2] export.php Illegal string offset 'method'

### ISSUE THREAD

**Issue by eschoeller** [Cacti/cacti#2143]
Certain graph templates fail to export correctly. I was having this problem in a Cacti instance running 1.0.38, and I upgraded it to 1.2.0 Beta 2 and still have the problem. I had someone else create these graph templates, and asked them to build them exactly the same way I had done. My graph template exports just fine, but the ones that were created by this individual are having a problem. The templates are working just fine and producing the correct graphs. But there is *something* out there that's tripping up the export. 
```
2018/11/02 20:25:19 - ERROR PHP WARNING: Illegal string offset 'method' in file: /cacti/cacti-1.2.0-beta2-test/lib/export.php  on line: 489
2018/11/02 20:25:19 - CMDPHP PHP ERROR WARNING Backtrace:  (/templates_export.php[33]:form_save(), /templates_export.php[58]:get_item_xml(), /lib/export.php[1116]:vdef_to_xml(), /lib/export.php[489]:CactiErrorHandler())
```
The resulting xml file has the following in it:
```
<b>Warning</b>:  Illegal string offset 'method' in <b>/cacti/cacti-1.2.0-beta2-test/lib/export.php</b> on line <b>489</b><br />
<br />
```

**Comment by netniV** [comment]
Ideally we could have used a dump prior to resetting the VDEF id.

**Comment by cigamit** [comment]
Good catch.  This should be resolved now.  Please update lib/export.php and test with the templates that have vdef's.

**Comment by eschoeller** [comment]
Well the templates which had VDEFs weren't supposed to have them. So I removed them and things worked out. I could create some templates with VDEFs and test it out, but I'm not sure I have the time to do that right now. I'd say close it if you think you nailed it.
